std/experimental/color/conv.d

```diff
@@ -0,0 +1,523 @@
+import std.experimental.color;
+import std.experimental.color.rgb;
+import std.experimental.color.xyz;
+
+import std.traits: isNumeric, isIntegral, isFloatingPoint, isSigned, TemplateOf;
+import std.typetuple: TypeTuple;
+
+
+/**
+Convert between color types.
+*/
+To convertColor(To, From)(From color) if(isColor!To && isColor!From)
+{
+    // no conversion is necessary
+    static if(is(To == From))
+        return color;
+
+    // *** XYZ is the root type ***
+    else static if(isXYZ!From && isXYZ!To)
+    {
+        alias F = To.ComponentType;
+        return To(F(color.X), F(color.Y), F(color.Z));
+    }
+
+    // following conversions come in triplets:
+    //   Type!U -> Type!V
+    //   Type -> Parent
+    //   Parent -> type
+
+    // *** RGB triplet ***
+    else static if(isRGB!From && isRGB!To)
+    {
+        alias ToType = To.ComponentType;
+        alias FromType = From.ComponentType;
+
+        auto src = color.tristimulusWithAlpha;
+
+        static if(false && From.colorSpace == To.colorSpace && isIntegral!FromType && FromType.sizeof <= 2 &&
+                    (From.linear != To.linear || !is(FromType == ToType)))
+        {
+            alias WorkType = WorkingType!(FromType, ToType);
+            enum NumValues = 1 << (FromType.sizeof*8);
+
+            // <= 16bit type conversion should use a look-up table
+            shared immutable ToType[NumValues] conversionTable = {
+                ToType[NumValues] table = void;
+                foreach(i; 0..NumValues)
+                {
+                    WorkType v = convertPixelType!WorkType(cast(FromType)i);
+                    static if(From.linear == false)
+                        v = toLinear!(From.colorSpace)(v);
+                    static if(To.linear == false)
+                        v = toGamma!(To.colorSpace)(v);
+                    table[i] = convertPixelType!ToType(v);
+                }
+                return table;
+            }();
+
+            static if(To.hasAlpha)
+                return To(conversionTable[src[0]], conversionTable[src[0]], conversionTable[src[1]], conversionTable[src[2]], convertPixelType!ToType(src[3]));
+            else
+                return To(conversionTable[src[0]], conversionTable[src[0]], conversionTable[src[1]], conversionTable[src[2]]);
+        }
+        else static if(From.colorSpace == To.colorSpace && From.linear == To.linear)
+        {
+            // color space is the same, just do type conversion
+            return To(convertPixelType!ToType(src[0]), convertPixelType!ToType(src[1]), convertPixelType!ToType(src[2]), convertPixelType!ToType(src[3]));
+        }
+        else
+        {
+            // unpack the working values
+            alias WorkType = WorkingType!(FromType, ToType);
+            WorkType r = convertPixelType!WorkType(src[0]);
+            WorkType g = convertPixelType!WorkType(src[1]);
+            WorkType b = convertPixelType!WorkType(src[2]);
+
+            static if(From.linear == false)
```

Comment:
Its been specifically suggested to use std.typecons.Flag here. Manu and I both disagree that it should be and we need to give reasons as to why.  It just happens to be that the reasons we think it shouldn't be used here is the reason we think it shouldn't be used at all.  And I'm happy to create a thread on the forum at a later date and ague for it not to be used at all.